arq/connections.py

```diff
@@ -38,6 +39,19 @@ class RedisSettings:
     sentinel: bool = False
     sentinel_master: str = 'mymaster'
 
+    @classmethod
+    def from_redis_url(cls, url: str):
+        """Parse a redis: connection URI, return a new
+        arq.connections.RedisSettings instance
+
+        :param url: The redis: url to parse
+        :return: :class:`arq.connections.RedisSettings` instance
+        """
+        address, options = parse_url(url)
```

Comment:
Does this check the URL scheme is correct?

Reply:
IIRC aioredis [`parse_url`](https://github.com/aio-libs/aioredis/blob/master/aioredis/util.py#L148-L149) does some error checking via asserts.

Reply:
Ye, aioredis has a lot of problems, I've started a new redis connection library, but it's far from finished.

I guess `assert` is okay for now here, very few people actually bother to disable debug in python.

Reply:
> I guess `assert` is okay for now here, very few people actually bother to disable debug in python.

Truth be told, as a user of arq, I wouldn't expect to perform validation on the redis connection URL string, other than to raise exceptions if grossly malformed (IOW can't parse in our helper code)

If anything, I assume it's the responsibility of underlying redis connection library, let it raise an exception if it wants to perform this level of validation.